<launch>

  <arg name="tuning_file_path" default="$(find-pkg-share paint_cloud)/config/tuning.yml" />
  <arg name="calibration_file_path" default="$(find-pkg-share paint_cloud)/config/calibration.yml" />
  <arg name="lidar_frame" default="lidar_top" />
  <arg name="base_frame" default="base_link" />
  <arg name="image_topic" default="/CF/compressed" />
  <arg name="cloud_topic" default="/sensing/lidar/compensated/pointcloud" />

  <node pkg="paint_cloud" exec="paint_rosbag" name="paint_rosbag" output="screen">
    <param name="bag_file_path" value="/run/user/1000/gvfs/smb-share:server=r4,share=aidata/VideoAi/ros_logs/foxy/nami/2023/2023-04/2023-04-11_DRedin/2023-04-11_DRedin_2_db3"/>
    <param name="painted_bag_directory" value="/home/m/projects/bag_files/2023-04-11_DRedin_2_db3"/>
    <param name="cloud_topic" value="$(var cloud_topic)"/>
    <param name="image_topic" value="$(var image_topic)"/>
  </node>
  
  <node pkg="segmentation_trt" exec="segmentation_trt_server" name="segmentation_trt_server" output="screen">
  </node>
  
  <node pkg="paint_cloud" exec="paint_cloud_server" name="paint_cloud_server" output="screen">
    <param name="tuning_file_path" value="$(var tuning_file_path)"/>
    <param name="calibration_file_path" value="$(var calibration_file_path)"/>
    <param name="lidar_frame" value="$(var lidar_frame)"/>
    <param name="base_frame" value="$(var base_frame)"/>
    <param name="image_topic" value="$(var image_topic)"/>
    <param name="cloud_topic" value="$(var cloud_topic)"/>
  </node>
</launch>
